ethereum/core/src/test/java/org/hyperledger/besu/ethereum/bonsai/worldview/BonsaiWorldStateTest.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.besu.ethereum.bonsai.worldview;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.datatypes.Hash;
+import org.hyperledger.besu.ethereum.bonsai.BonsaiValue;
+import org.hyperledger.besu.ethereum.bonsai.cache.CachedMerkleTrieLoader;
+import org.hyperledger.besu.ethereum.bonsai.storage.BonsaiWorldStateKeyValueStorage;
+import org.hyperledger.besu.ethereum.bonsai.storage.BonsaiWorldStateLayerStorage;
+import org.hyperledger.besu.ethereum.bonsai.trielog.TrieLogManager;
+import org.hyperledger.besu.ethereum.core.InMemoryKeyValueStorageProvider;
+import org.hyperledger.besu.metrics.noop.NoOpMetricsSystem;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class BonsaiWorldStateTest {
+  @Mock TrieLogManager trieLogManager;
+  @Mock CachedMerkleTrieLoader cachedMerkleTrieLoader;
+  @Mock BonsaiWorldStateUpdateAccumulator bonsaiWorldStateUpdateAccumulator;
+  @Mock BonsaiWorldStateKeyValueStorage.BonsaiUpdater bonsaiUpdater;
+
+  private static final Bytes CODE = Bytes.of(10);
+  private static final Hash CODE_HASH = Hash.hash(CODE);
+  private static final Hash ACCOUNT_HASH = Hash.hash(Address.ZERO);
+  private static final Address ACCOUNT = Address.ZERO;
+
+  final BonsaiWorldState worldState =
+      new BonsaiWorldState(
+          new BonsaiWorldStateLayerStorage(
+              new BonsaiWorldStateKeyValueStorage(
+                  new InMemoryKeyValueStorageProvider(), new NoOpMetricsSystem())),
+          cachedMerkleTrieLoader,
+          trieLogManager);
+
+  @ParameterizedTest
+  @MethodSource("priorAndUpdatedEmptyAndNullBytes")
+  void doesNothingWhenMarkedAsDeletedButAlreadyDeleted(final Bytes prior, final Bytes updated) {
```

Comment:
nit: test names don't really indicate it's related to updating the code storage.

Reply:
Good point missed will fix those names up